src/code/utils/translate.coffee

```diff
@@ -5,7 +5,8 @@ varRegExp = /%\{\s*([^}\s]*)\s*\}/g
 
 translate = (key, vars={}, lang=defaultLang) ->
   translation = translations[lang]?[key] or key
-  translation.replace varRegExp, (match, key) ->
-    if vars.hasOwnProperty key then vars[key] else "'** UKNOWN KEY: #{key} **"
+  if translation
```

Comment:
I actually encountered this at some point during development. I don't recall the specific circumstances but it was probably something like the client was using an ill-formed if expression which resulted in passing undefined as the key. It didn't seem appropriate for the translate function to crash at that point, although there may be a better return value in that scenario than the current undefined.